### PR TITLE
added image credentials info to usage/overview/image

### PR DESCRIPTION
**Pull request by Zhomart**
From Brad's answer on Gitter

```
>> I don't have it documented yet, unfortunately, but there are reserved environment variable names that if configured, will be used to pull the image. You can sort of see the logic here https://github.com/drone/drone/blob/master/yaml/transform/secret.go#L25:L31
for example drone secret add --image=gcr.io/me/repo:tag my/repo REGISTRY_USERNAME _json_key
will instruct drone to pull gcr.io/me/repo:tag using username _json_key
```

### CODE REVIEW

**Comment by gtaylor** [comment]
@bradrydzewski I might defer to you on this one.

**Comment by tboerger** [comment]
> @bradrydzewski I might defer to you on this one.

This is generally correct, take a look at https://github.com/drone/drone/blob/master/yaml/transform/secret.go#L19-L39 for reference

File: content/usage/overview.md [hunk 1]
```diff
@@ -70,6 +70,15 @@ pipeline:
     image: golang:1.6
 ```
 
+For private repositories credentials can be configured by secrets
```
Comment (tboerger) [inline content/usage/overview.md]: I would rephrase it more like that:

For private repositories you have to provide secrets to the docker daemon, valid secrets are `REGISTRY_USERNAME`, `REGISTRY_PASSWORD` and `REGISTRY_EMAIL`.

You can set these secrets using the drone-cli:

```
drone secret add --image=my.registry.com:5000/my/repo:tag \
    octocat/hello-world REGISTRY_USERNAME octocat

drone secret add --image=my.registry.com:5000/my/repo:tag \
    octocat/hello-world REGISTRY_PASSWORD mypassword

drone secret add --image=my.registry.com:5000/my/repo:tag \
    octocat/hello-world REGISTRY_EMAIL octocat@github.com
```

**Comment by tboerger** [comment]
We can close this PR, it's already part of https://github.com/drone/docs/blob/release/0.5/content/usage/services.md#authorization
